File: tools/misc/phases/post-build.rb
require_relative '../../misc/inject-script-loaders/manifest'
require_relative '../../misc/launchDev/writeVersion'
require_relative '../../misc/launchDev/savePrefs'

def log_info(msg) = puts "[INFO] #{msg}"
def log_success(msg) = puts "[SUCCESS] #{msg}"

def write_buildid2(bin_path, buildid)
  File.write(File.join(bin_path, 'buildid2.txt'), buildid)
end

def run_post_build_phase(is_dev = false)
  log_info '🔨 Starting post-build phase...'
  bin_path = File.expand_path('_dist/bin/noraneko', Dir.pwd)
  log_info '📋 Injecting manifest...'
  inject_manifest(bin_path, is_dev ? 'dev' : 'prod')
  log_info '🌐 Injecting XHTML...'
  # Skipped: injectXHTML/injectXHTMLDev (xhtml.ts is excluded)
  log_info '🆔 Writing build ID...'
  write_buildid2(bin_path, 'nora-build-id')
  if is_dev
    log_info '🚀 Setting up development environment...'
    gen_version
    save_prefs_for_profile(File.expand_path('_dist/profile/test', Dir.pwd))
  end
  log_success 'Post-build phase completed'
end

run_post_build_phase(ARGV[0] == 'dev') if $PROGRAM_NAME == __FILE__

File: tools/misc/phases/post-build.ts
import { log } from "../../utils/logger.ts";
import { injectManifest } from "../inject-script-loaders/manifest.ts";
import { injectXHTML, injectXHTMLDev } from "../inject-script-loaders/xhtml.ts";
import { writeBuildid2 } from "../update/buildid2.ts";
import { genVersion } from "../launchDev/writeVersion.ts";
import { savePrefsForProfile } from "../launchDev/savePrefs.ts";
import { resolve } from "@std/path";
import { resolveFromRoot } from "../../utils/utils.ts";

export async function runPostBuildPhase(isDev = false): Promise<void> {
  log.info("🔨 Starting post-build phase...");

  const binPath = resolveFromRoot("_dist/bin/noraneko");

  log.info("📋 Injecting manifest...");
  await injectManifest(binPath, isDev ? "dev" : "prod");

  log.info("🌐 Injecting XHTML...");
  await (isDev ? injectXHTMLDev(binPath) : injectXHTML(binPath));

  log.info("🆔 Writing build ID...");
  await writeBuildid2(binPath, "nora-build-id");

  if (isDev) {
    log.info("🚀 Setting up development environment...");
    await genVersion();
    await savePrefsForProfile(resolve(resolveFromRoot("_dist/profile/test")));
  }

  log.success("Post-build phase completed");
}
